CPS-????/README.md

```diff
@@ -0,0 +1,81 @@
+
+## Problem
+Imagine a scenario where Alice would like to trustlessly swap assets with Bob. An atomic swap contract is used and any UTxOs that are to be swapped contain an inline datum of an asking price. Here is a table to represent all of Alice's available UTxOs to swap (at the atomic swap script address):
+
+|UTxO ID| Asking Price|Amount|
+|--|--|--|
+| 1 | 1.5 DUST/ADA | 10 ADA |
+| 2 | 2 DUST/ADA | 20 ADA |
+| 3 | 1 DUST/ADA | 5 ADA |
+
+If Bob wants to swap his DUST for Alice's ADA, he will need to satisfy Alice's asking price. As an example, if he wants to take all 10 ADA from UTxO 1, he will need to deposit 15 DUST to Alice's atomic swap address in the same transaction that consumes UTxO 1.
+
+But what if he wants to swap his DUST for 25 ADA? That would require using at least two of Alice's UTxOs. Doing this is trivial for the atomic swap contract because it validates based off of the transaction as a whole: 
+
+``` Txt
+The value being deposited to the swap address must be >= 
+  the value withdrawn from the swap address * 
+  the weighted average asking price of all UTxOs being spent from the swap address
+```
+
+This brings us to the problem: **there is currently no way to efficiently use a plutus spending script that validates based off of the transaction as a whole**. (These scripts will be called *transaction level spending scripts* for the rest of this CPS). With the current design of Cardano, if Bob consumed UTxO 2 and UTxO 3 in the same transaction to satisfy his desire for 25 ADA, the atomic swap contract will be executed twice. Since the transaction context doesn't change between executions, the second execution is completely redundant.
+
+Any time a developer creates a transaction level spending script, it will be redundantly executed for every additional UTxO consumed from the corresponding script address. These redundant executions are a waste of scarce resources and result in much higher fees for end users. Not only that, but the inability to efficiently use these transaction level spending scripts significantly handicaps the use of the eUTxO model. The eUTxO model already allows for creating such spending scripts; developers are just unable to properly use them.
+
+### The Current Alternative
+The alternative involves taking advantage of the fact that staking scripts do not suffer from the same limitation as spending scripts.
```

Comment:
I think the explanation for why this works is wrong. Staking scripts _do_ get information about what script is running and _do_ get redeemers. 

The reason this works is because you get all your spending scripts to coordinate on delegating to a _single_ other script execution. You could do exactly this trick with another spending script: spend an additional UTXO locked with a script that does the "real work", and have all the other UTXO spending scripts just assert that that script is being spent in the transaction. Same effect!

Reply:
Your reason doesn't not account for everything in the actual implementation. Using another spending script was discussed [here](https://github.com/cardano-foundation/CIPs/pull/418#issuecomment-1367309018). It is not a viable solution because it creates a bottleneck problem: you are only as concurrent as the number of utxos available at the spending script being delegated to. This bottleneck problem does not exist for staking scripts since no utxos are needed to force execution of the script.

Reply:
The point is not that it's a workaround, my point is that your explanation for why it works is wrong.